### ISSUE TITLE
java.lang.NullPointerException when importing csv file into Deposits account

### ISSUE THREAD

**Issue by litebito** [portfolio-performance/portfolio#1971]
**Describe the bug**
Since today, it seems I'm no longer able to import CSV files.
Even files which I could import earlier without issues, now give the same problem : **java.lang.NullPointerException**
So I'm blocked from continuing using PP at the moment.

**To Reproduce**
1. Go to 'File, Import, CSV files'
2. Select my CSV file
3. Change encoding to UTF-8
4. Ignore all columns except Date, Value, Type, Note in the CSV file
5. Set the date column to match the format DD/MM/YYYY
6. Click on 'Next, Finish'
Then I get the error.
if I then click "Back" I get the error again

```
`Thu Jan 14 00:23:01 CET 2021
Internal Error

java.lang.NullPointerException
	at name.abuchen.portfolio.ui.wizards.datatransfer.ReviewExtractedItemsPage.afterPage(ReviewExtractedItemsPage.java:704)
	at name.abuchen.portfolio.ui.wizards.datatransfer.CSVImportWizard.performFinish(CSVImportWizard.java:163)
	at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:832)
	at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:472)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:84)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4209)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1037)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3626)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at name.abuchen.portfolio.ui.handlers.ImportCSVHandler.runImport(ImportCSVHandler.java:87)
	at name.abuchen.portfolio.ui.handlers.ImportCSVHandler.lambda$0(ImportCSVHandler.java:46)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at name.abuchen.portfolio.ui.handlers.ImportCSVHandler.execute(ImportCSVHandler.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:319)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:253)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4209)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1037)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3626)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Application.start(E4Application.java:166)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1461)

`
```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version 0.50.1

**Comment by buchen** [comment]
Hi @litebito,

this stack trace seems to be the **second** error message.

Can you post the **first** error message as well (go the error log -> double click -> copy stack trace).

**Comment by litebito** [comment]
Hi,

I only always see 1 error in the error log when I try.

So I tried again today 1 time, this is the "copy" of the whole trace I can see

```
Sun Jan 17 15:09:33 CET 2021
Internal Error
java.lang.NullPointerException
	at name.abuchen.portfolio.ui.wizards.datatransfer.ReviewExtractedItemsPage.afterPage(ReviewExtractedItemsPage.java:704)
	at name.abuchen.portfolio.ui.wizards.datatransfer.CSVImportWizard.performFinish(CSVImportWizard.java:163)
	at org.eclipse.jface.wizard.WizardDialog.finishPressed(WizardDialog.java:832)
	at org.eclipse.jface.wizard.WizardDialog.buttonPressed(WizardDialog.java:472)
	at org.eclipse.jface.dialogs.Dialog.lambda$0(Dialog.java:619)
	at org.eclipse.swt.events.SelectionListener$1.widgetSelected(SelectionListener.java:84)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:252)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4209)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1037)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3626)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at name.abuchen.portfolio.ui.handlers.ImportCSVHandler.runImport(ImportCSVHandler.java:87)
	at name.abuchen.portfolio.ui.handlers.ImportCSVHandler.lambda$0(ImportCSVHandler.java:46)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at name.abuchen.portfolio.ui.handlers.ImportCSVHandler.execute(ImportCSVHandler.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invokeUsingClass(InjectorImpl.java:319)
	at org.eclipse.e4.core.internal.di.InjectorImpl.invoke(InjectorImpl.java:253)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.invoke(ContextInjectionFactory.java:173)
	at org.eclipse.e4.core.commands.internal.HandlerServiceHandler.execute(HandlerServiceHandler.java:156)
	at org.eclipse.core.commands.Command.executeWithChecks(Command.java:488)
	at org.eclipse.core.commands.ParameterizedCommand.executeWithChecks(ParameterizedCommand.java:487)
	at org.eclipse.e4.core.commands.internal.HandlerServiceImpl.executeHandler(HandlerServiceImpl.java:213)
	at org.eclipse.e4.ui.workbench.renderers.swt.HandledContributionItem.executeItem(HandledContributionItem.java:438)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.handleWidgetSelection(AbstractContributionItem.java:449)
	at org.eclipse.e4.ui.workbench.renderers.swt.AbstractContributionItem.lambda$2(AbstractContributionItem.java:475)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:89)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4209)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1037)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4026)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3626)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1157)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Application.start(E4Application.java:166)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:653)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:590)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1461)


```



the first few lines of the csv:
```
Date,Value,Type,Note
7/02/2016,1.4,Interest,The Agent Crowd 
7/03/2016,1.4,Interest,The Agent Crowd 
7/04/2016,1.4,Interest,The Agent Crowd 
7/05/2016,1.4,Interest,The Agent Crowd 
```

Screenshots:
https://imgur.com/Uwbaagl
https://imgur.com/7vZTAT3

**Comment by buchen** [comment]
Thanks. I kind of can reproduce it but only if I have a file that has **no** cash accounts or security accounts.

Can you send me your file to *portfolio dot performance dot help at gmail dot com*?

**Comment by litebito** [comment]
I think I just found the problem.
When starting with the file, it makes you create a Securities Account
Now because I wanted to only for my P2P platforms, a few days ago, I removed the "Temp" Securities Account. Created in the beginning. 
Based on your remark above, I checked and during the import, we have to specify both a cash account and securities account. But with me the securities account is empty (because I removed them all). 
I went back, and added a security account again (named "General"), and coupled it with an empty cash account "General" I created (but not plan to use)

I tried the import again, and now I select the named cash account I want to use ("looknfin" in this case) and select the "general" securities account. Now the import seems to work fine, I got 1 more time an error, cleared the looknfin account from all transactions, tried import again, now it works without errors.

So basically, even if you don't use it, you always need to have a securities account specified

Does that sound correct?

**Comment by mierin12** [comment]
Yes minimum of 1 cash account and 1 securities account are required, as you can import not only cash account transactions but also portfolio transactions by CSV.

Fixed in released [release 0.50.2](https://github.com/portfolio-performance/portfolio/releases/tag/0.50.2) with https://github.com/portfolio-performance/portfolio/commit/d12c3e118fe722f558d2d362489d624ac68d1831
